javascript-bank/transaction.js

```diff
@@ -0,0 +1,8 @@
+/* exported Transaction */
+
+function Transaction(inType, inAmount) {
+    this.type = inType,
+    this.amount = inAmount
+
+    return this.type, this.amount
```

Comment:
The only reason this returns a `Transaction` instance and not the `amount` property of `this` is that the `amount` property contains a number instead of an object, which causes the `new` operator to return `this` instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new#description

As it's written, the `return` statement will confuse a lot of people who read it, since it seems like something other than `this`. Either explicitly return `this`, or leave off the `return` statement entirely.

---

![image](https://user-images.githubusercontent.com/16143771/123364279-ffaf0380-d528-11eb-8b55-ba39596a287d.png)


---

Reply:
removed the return statement from function definition